The repo fails to build when git source control information isn't available

Command: "./build.sh" on root
I didn't use https://github.com/dotnet/source-build

Log (copied from Bash): pastebin.com/6MEtHDcq

```
/root/.nuget/packages/microsoft.build.tasks.git/1.1.0-beta-20206-02/build/Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory '/home/osboxes/Desktop/runtime-5.0.2/tools-local/tasks/installer.tasks'. [/home/osboxes/Desktop/runtime-5.0.2/tools-local/tasks/installer.tasks/installer.tasks.csproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.1.0-beta-20206-02/build/Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory '/home/osboxes/Desktop/runtime-5.0.2/tools-local/tasks/installer.tasks'. [/home/osboxes/Desktop/runtime-5.0.2/tools-local/tasks/installer.tasks/installer.tasks.csproj]
/root/.nuget/packages/microsoft.sourcelink.common/1.1.0-beta-20206-02/build/Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [/home/osboxes/Desktop/runtime-5.0.2/tools-local/tasks/installer.tasks/installer.tasks.csproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.1.0-beta-20206-02/build/Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory '/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/src/System.Private.CoreLib'. [/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj]
/root/.nuget/packages/microsoft.build.tasks.git/1.1.0-beta-20206-02/build/Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory '/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/src/System.Private.CoreLib'. [/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj]
/root/.nuget/packages/microsoft.sourcelink.common/1.1.0-beta-20206-02/build/Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/src/System.Private.CoreLib/System.Private.CoreLib.csproj]
/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/runtime.proj(32,5): error MSB3073: The command ""/home/osboxes/Desktop/runtime-5.0.2/src/coreclr/build-runtime.sh" -x64 -debug -os Linux" exited with code 1.
    0 Warning(s)
    7 Error(s)
```

Reprod locally by checking out the repo and deleteing the ".git" folder. Here's what I get:

```
  C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\eng'. [C:\git\runtime6\eng\empty.csproj]
  C:\Users\vihofer\.nuget\packages\microsoft.dotnet.arcade.sdk\6.0.0-beta.21062.10\tools\Version.targets(53,5): error : SourceRevisionId is not set, which means the SourceLink targets are not included in the build. Those are needed to produce a correct sha for our build outputs. [C:\git\runtime6\eng\empty.csproj]
  Build failed with exit code 1. Check errors above.
  See log: "C:\git\runtime6\src\coreclr\..\..\artifacts\log\Debug\GenerateVersionHeaders_windows__x64__Debug.binlog"
  BUILD: Error: Failed to generate version headers.
C:\git\runtime6\src\coreclr\runtime.proj(42,5): error MSB3073: The command ""C:\git\runtime6\src\coreclr\build-runtime.cmd" -x64 -debug" exited with code 1.

Build FAILED.

C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\tasks\installer.tasks'. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\tasks\installer.tasks'. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.sourcelink.common\1.1.0-beta-20206-02\build\Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\tasks\installer.tasks'. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\tasks\installer.tasks'. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.sourcelink.common\1.1.0-beta-20206-02\build\Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [C:\git\runtime6\src\tasks\installer.tasks\installer.tasks.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\System.Private.CoreLib'. [C:\git\runtime6\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\System.Private.CoreLib'. [C:\git\runtime6\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.sourcelink.common\1.1.0-beta-20206-02\build\Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [C:\git\runtime6\src\coreclr\System.Private.CoreLib\System.Private.CoreLib.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\ToolBox\SOS\DIALib'. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DIALib\DIALib.ilproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\ToolBox\SOS\DIALib'. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DIALib\DIALib.ilproj]
C:\Users\vihofer\.nuget\packages\microsoft.sourcelink.common\1.1.0-beta-20206-02\build\Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DIALib\DIALib.ilproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(24,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\ToolBox\SOS\DacTableGen'. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DacTableGen\DacTableGen.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.build.tasks.git\1.1.0-beta-20206-02\build\Microsoft.Build.Tasks.Git.targets(47,5): error : Unable to locate repository with working directory that contains directory 'C:\git\runtime6\src\coreclr\ToolBox\SOS\DacTableGen'. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DacTableGen\DacTableGen.csproj]
C:\Users\vihofer\.nuget\packages\microsoft.sourcelink.common\1.1.0-beta-20206-02\build\Microsoft.SourceLink.Common.targets(52,5): error : Source control information is not available - the generated source link is empty. [C:\git\runtime6\src\coreclr\ToolBox\SOS\DacTableGen\DacTableGen.csproj]
C:\git\runtime6\src\coreclr\runtime.proj(42,5): error MSB3073: The command ""C:\git\runtime6\src\coreclr\build-runtime.cmd" -x64 -debug" exited with code 1.
```

Apparently coreclr intentionally fails the build (in empty.csproj) if source control information isn't available. I'm unsure why this is the case. @tmat is it intentional that when referencing the source link packages that the build for managed project fails like in the logs above? I assume we are doing something wrong in coreclr.

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -193,6 +193,10 @@
     <RestoreUseStaticGraphEvaluation>true</RestoreUseStaticGraphEvaluation>
     <!-- Turn off end of life target framework checks as we intentionally build older .NETCoreApp configurations. -->
     <CheckEolTargetFramework>false</CheckEolTargetFramework>
+    <!-- Disable source link when building locally. -->
+    <DisableSourceLink Condition="'$(DisableSourceLink)' == '' and
+                                  '$(ContinuousIntegrationBuild)' != 'true' and
+                                  '$(OfficialBuildId)' == ''">false</DisableSourceLink>
   </PropertyGroup>
 
   <!-- RepositoryEngineeringDir isn't set when Installer tests import this file. -->

--- a/src/installer/Directory.Build.props
+++ b/src/installer/Directory.Build.props
@@ -28,12 +28,6 @@
     <Platform Condition="'$(Platform)' == ''">$(TargetArchitecture)</Platform>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(DisableSourceLink)' == 'true'">
-    <EnableSourceLink>false</EnableSourceLink>
-    <EnableSourceControlManagerQueries>false</EnableSourceControlManagerQueries>
-    <DeterministicSourcePaths>false</DeterministicSourcePaths>
-  </PropertyGroup>
-
   <!-- Set up Default symbol and optimization for Configuration -->
   <PropertyGroup Condition="'$(Configuration)'=='Debug'">
     <DefineConstants>$(DefineConstants),DEBUG,TRACE</DefineConstants>

--- a/src/libraries/Directory.Build.props
+++ b/src/libraries/Directory.Build.props
@@ -95,9 +95,6 @@
     <StrongNameKeyId>Open</StrongNameKeyId>
     <!-- Microsoft.Extensions projects have a separate StrongNameKeyId -->
     <StrongNameKeyId Condition="$(MSBuildProjectName.StartsWith('Microsoft.Extensions.'))">MicrosoftAspNetCore</StrongNameKeyId>
-
-    <!-- Workaround for codecov issue https://github.com/tonerdo/coverlet/issues/312 -->
-    <EnableSourceLink Condition="'$(ContinuousIntegrationBuild)' != 'true' and '$(OfficialBuildId)' == ''">false</EnableSourceLink>
   </PropertyGroup>
 
   <!-- Disable some standard properties for building our projects -->

